src/random_object.cpp

```diff
@@ -32,7 +32,7 @@ Value RandomObject::bytes(Env *env, Value size) {
         env->raise("ArgumentError", "negative string size (or size too big)");
 
     TM::String output(static_cast<size_t>(isize), '\0');
-    std::uniform_int_distribution<char> random_number {};
+    std::uniform_int_distribution<nat_int_t> random_number {};
     for (nat_int_t i = 0; i < isize; i++)
         output[i] = random_number(*m_generator);
```

Comment:
More a future work, but it looks like we could this more efficiently. This code requests 8 bytes of random data (assuming a 64 bit arch), and throws away 7 of them.

Reply:
Are you suggesting using a different integer type? Which type would be best to use here?

Reply:
I was thinking the other way around: we get 8 bytes, so let's use all 8 bytes. So something like this:
```c++
for (nat_int_t i = 0; i < isize; i += 8) {
  const auto rand = random_number(*m_generator);
  output[i] = rand >> 56;
  output[i + 1] = rand >> 48
  // ....
  output[i + 7] = rand
}
```
But using `sizeof(nat_int_t)` instead of the hardcoded 8, no manual loop unrolling and actually supporting sizes that are not multiples of 8.

Reply:
yeah let's ship it 😄 🚢

Reply:
@herwinw Ah ok, less wasteful certainly! The reddit thread I linked suggests the approach is inefficient but I have neither the necessary understanding or the time for re-implementing it right now 😊